engine/base/api/src/main/scala/pl/touk/nussknacker/engine/baseengine/api/runtimecontext/EngineRuntimeContextPreparer.scala

```diff
@@ -0,0 +1,20 @@
+package pl.touk.nussknacker.engine.baseengine.api.runtimecontext
+
+import pl.touk.nussknacker.engine.api.runtimecontext.EngineRuntimeContext
+import pl.touk.nussknacker.engine.util.metrics.{MetricsProvider, NoOpMetricsProvider}
+
+object EngineRuntimeContextPreparer {
+
+  val forTest = new EngineRuntimeContextPreparer(_ => new NoOpMetricsProvider with AutoCloseable {
+    override def close(): Unit = {}
+  })
+
+}
+
+class EngineRuntimeContextPreparer(metricRegistryForScenario: String => MetricsProvider with AutoCloseable) {
```

Comment:
IMO we won't need this class if `MetricProvider` will be a "real" metric mechanism provider. Currently it join Flink's `MetricGroup` which is responsible for providing metrics that are in scope of Task/Slot metric group and is a part of operators (our components) API and Flink's `MetricReporter` API which can report metric for any identifier. I wonder if we can talk about how it could be designed to be more clear? Maybe using similar names like in Flink will help us?

Reply:
Changed name to point that it is only for particular scenario